cinema/serializers.py

```diff
@@ -68,13 +79,81 @@ class Meta:
             "movie_title",
             "cinema_hall_name",
             "cinema_hall_capacity",
+            "tickets_available",
         )
 
 
+class TicketSerializer(serializers.ModelSerializer):
+    movie_session = MovieSessionSerializer()
```

Comment:
The `movie_session` field in `TicketSerializer` is using `MovieSessionSerializer`, which may not provide the full `cinema_hall` object required in the `validate` method. Consider using a primary key related field or ensuring the full object is available.